src/test/java/com/marklogic/spark/reader/PushDownGroupByAvgTest.java

```diff
@@ -0,0 +1,63 @@
+package com.marklogic.spark.reader;
+
+import com.marklogic.spark.Options;
+import org.apache.spark.sql.Dataset;
+import org.apache.spark.sql.Row;
+import org.junit.jupiter.api.Test;
+
+import java.util.List;
+
+import static org.apache.spark.sql.functions.avg;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+public class PushDownGroupByAvgTest extends AbstractPushDownTest {
+
+    @Test
+    void groupByAvg() {
+        verifyRows(
+            "avg(LuckyNumber)",
+            newDefaultReader()
+                .option(Options.READ_OPTIC_QUERY, QUERY_WITH_NO_QUALIFIER)
+                .load()
+                .groupBy("CitationID")
+                .avg("LuckyNumber")
+                .orderBy("CitationID")
+        );
+    }
+
+    @Test
+    void aggAvg() {
+        verifyRows(
+            "avg(LuckyNumber)",
+            newDefaultReader()
+                .option(Options.READ_OPTIC_QUERY, QUERY_WITH_NO_QUALIFIER)
+                .load()
+                .groupBy("CitationID")
+                .agg(avg("LuckyNumber"))
+                .orderBy("CitationID")
+        );
+    }
```

Comment:
For the previous two tests: they produce the exact same output? What does .agg() do?

Reply:
I believe `agg()` is typically used to combine multiple aggregations, but it works with a single one. I added a test like that for each aggregate function to ensure that they behave the same way.